articles/node_lifecycle.md

```diff
@@ -0,0 +1,176 @@
+
+To shut down a node, it must first be deactivated. This will cause the `onDeactivate` method to be called, allowing the node to clean up any resources it only holds while active.
+
+If an error that cannot be handled by the node/system occurs in this state, the transition to the `fatal error` state is taken.
+
+### Fatal Error state
+
+Known errors that can occur in a node should be considered designed behaviour of the system and so should be handled by the node where the error occurred and, if necessary, the surrounding nodes. The reason for this is that the system knows best how to handle errors that occur in its behaviour.
+
+Any fatal errors (those which the node/system was not expecting and does not know how to handle) will cause a transition to the `fatal error` state. In this state, because the node does not know what to do in the presence of the error, it is not provided any execution time. Arriving data is not processed, service requests are not responded to, and no output is produced.
+
+On entering the `fatal error` state, the node's `onError` method must be executed. This provides the node with a chance to clean up resources it was using.
+
+When a node is in this state, roslaunch may choose to reactivate the node and hope the error has gone away (accomplished by clearing the error and then going through the standard activation process), restart the node (and hope the error does not repeat), or it may choose to replace it with an alternative implementation that provides the same functionality. Restarting or replacing the node can be accomplished destroying it and creating a new instance of its replacement.
+
+### Destroyed state
+
+This is the state that a node goes through on its way to no longer existing.
+
+A node must be deactivated before it can be destroyed.
+
+On entering this state, the node's `onDestroyed` method must be run. This is the node's last chance to clean up.
+
+Implementation note: In object-oriented languages, the onDestroyed method can be mapped to the class destructor.
+
+
+
+
+## Interface
+
+A node must provide the messages and services defined in this section to qualify as a managed node. These services define the external interface of a managed node as seen by tools that perform the managing. As such, it should be possible to consider any object that provides these services correctly to be a managed node.
+
+Conversely, any object that provides these services but does not behave in the way defined in the life cycle state machine is malformed.
+
+These services may be provided via attributes and method calls (for local management) or via ROS messages and topics/services (for remote management). In the case of providing a ROS middleware interface, specific topics must be used, and they should be placed in a suitable namespace.
+
+### Initialisation
+
+The following service should be called to initialise the node. This service must be named `initialise`.
```

Comment:
I actually thought about this ... using services to manage the node's lifecycle, while stating that an unactive node cannot respond to services is inconsistent.